### PR TITLE
LPS-177564 Navigate to FDS Views page on the 'view' action of FDS Entry

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-177564 Navigate to FDS Views page on the 'view' action of FDS Entry](https://issues.liferay.com/browse/LPS-177564)
- [Figma mockup](https://www.figma.com/file/VTOLTBFe99y9kA7AY3TcNR/lps-175276-design-app-to-manage-datasets?node-id=937-117695&t=gNfcz6mQplE92gsc-0)

### What is the goal of this PR?

This is the first PR of the story to manage FDS Views. We are adding navigation from FDS Entry action to the placeholder of FDS Views page.

Notes:
- The changes in FDS Views Manager are minimal, basically a sample of frontend setup of actions, with a placeholder Views page. I'll send FDS Views persistence and CRUD handling in a separate PR.
- Setting of `onClick` in actions was bugged, this is a part of our discovery of issues when using FDS directly as a React component. In the PR we fixing this, but also taking the opportunity to massively simplify handling of actions in FDS. @carloslancha I had something like this in mind when we discussed exposing of FDS context in custom actions. There is now a single point of placing exposed values, in `exposedProps` const. There are for sure other ways to simplify code, but I had to stop myself :slightly_smiling_face: 
  - As a part of the changes, we are fixing accessibility `aria-label` setting, fixing `PropTypes` setting, and removing unused `size` setting.
  - It may be that I missed some edge case, but I did test several places in commerce and FDS sample module.
- Inline icon for action is temporary, until we add other actions. Then, it will become a dropdown, as in mockups. cc @ugeortiz

### What does it look like?

https://user-images.githubusercontent.com/5435511/223713113-aa76822e-c2b1-4cca-a13b-d9cebc59d7a6.mp4

### Steps to reproduce
1. Add `feature.flag.LPS-164563=true` to `portal-ext.properties`
2. Open Global Menu > Control Panel > Object > Datasets

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js [hunk 1]
```diff
@@ -190,47 +96,103 @@ function Actions({actions, itemData, itemId, menuActive, onMenuActiveChange}) {
...
+	const handleClick = ({action, closeMenu, event}) => {
+		const {data, href, method, onClick, target} = action;
+
+		const {
+			confirmationMessage,
+			errorMessage,
+			status,
+			successMessage,
+			title,
+		} = data ?? {};
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js:108]: Neat trick! I wonder how the rest of the code functions if `data` is falsy, do we need any sanity checks?

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js [hunk 2]
```diff
@@ -190,47 +96,103 @@ function Actions({actions, itemData, itemId, menuActive, onMenuActiveChange}) {
...
+						title,
+						url,
+					});
+				}
+			}
+			else if (target === 'sidePanel') {
+				event.preventDefault();
+
+				highlightItems([itemId]);
+				openSidePanel({
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js:131]: Empty line or am I trolling?

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/ActionsDropdown.js [hunk 3]
```diff
@@ -88,11 +75,10 @@ function ActionsDropdown({
 
 	const isMounted = useIsMounted();
 
-	const editModeActive = !!frontendDataSetContext.itemsChanges[itemId];
+	const editModeActive = !!itemsChanges[itemId];
 	const itemChanges =
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/ActionsDropdown.js:79]: Probably empty line missing. I remember the rule empty line between declaration and usage

**Review by kresimir-coko — CHANGES_REQUESTED** [review]
Small things but looking great

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/ActionsDropdown.js (hunk 3 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/ActionsDropdown.js:79]: That's a rule in Java, currently optional in JS.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx [hunk 4]
```diff
@@ -430,9 +432,25 @@ const FDSEntries = ({
 		],
 	};
 
+	const onViewClick = ({itemData}: any) => {
```
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx:435]: Oops! That should have been temporary, to get TS out of the way until logic is done :sweat_smile:

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js:108]: There is one other place where `data` is used, and we do have sanity check:
 ```js
executeAsyncItemAction({
	...
	method: method ?? data?.method,
	...
});
```
This is one the things that can be simplified, there is no point in `method` being defined in two places in API. But, I'd put that outside of scope of this PR. We would need to check all usages, possibly change some of them.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js (hunk 2 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js:131]: /giphy they see me trolling

**Comment by markocikos** [comment]
@kresimir-coko Force-pushed! Changes:
- Fixed `dataType`, based on https://github.com/liferay-frontend/liferay-portal/pull/3103#discussion_r1129575381
- SFd, based on trolling
- Moved `itemsActions` definition in its right place, as we discussed in Slack
- Fixed unconditional prevention of default, that caused CI error in https://github.com/liferay-frontend/liferay-portal/pull/3103#issuecomment-1460455812

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx [hunk 5]
```diff
@@ -463,6 +472,13 @@ const FDSEntries = ({
 					provider: ProviderRenderer,
 				}}
 				id={`${namespace}FDSEntries`}
+				itemsActions={[
+					{
+						icon: 'view',
+						label: Liferay.Language.get('view'),
+						onClick: onViewClick,
+					},
+				]}
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx:481]: I see you decided to leave it here in `views`.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx [hunk 6]
```diff
@@ -430,6 +430,15 @@ const FDSEntries = ({
 		],
 	};
 
+	const onViewClick = ({itemData}: {itemData: FDSEntry}) => {
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx:433]: Can't this just be `const onViewClick = ({itemData}: FDSEntry) => {` ?

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts [hunk 7]
```diff
@@ -69,6 +69,7 @@ type TItemsActions = {
 	icon?: string;
 	id?: string;
 	label?: string;
+	onClick?: Function;
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts:72]: So you added `onClick` to `TItemsActions`, what do I do if I need a small modal as a `target`? Should I add the type as a part of my changes?

**Review by kresimir-coko — APPROVED** [review]
Tiny questions

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx (hunk 6 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx:433]: That would mean that `itemData` is a property of `FDSEntry`. TS Error:
```ts
Property 'itemData' does not exist on type 'FDSEntry'.
```

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts (hunk 7 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts:72]: If you need it, yes. We don't need to be comprehensive about types, adding only used properties. It's quite likely that we have unused ones. We can make a comprehensive sweep when we migrate FDS to TS. 

On a sidenote, you should probably not use modal target, as this uses custom FDS modal. We intend to replace it with DXP standard `openModal`. In general, `action.target` is not a great API. The main reason for it is to have configuration in Java. You set a keyword, `target`, and this determines hardcoded frontend behavior. We don't need to limit ourselves in that way when using `FrontendDataSet` React component directly.

Comment (kresimir-coko) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts:72]: Okay, so I guess something like an onClick that opens our frontend-js-web modal would be the preferred route?

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx (hunk 6 above)
Comment (kresimir-coko) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx:433]: Ah, okay, still rusty with TS

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts (hunk 7 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts:72]: yep :+1:

**Comment by markocikos** [comment]
ci:forward
